mbed_devices/_internal/utils/python_helpers.py

```diff
@@ -0,0 +1,23 @@
+"""Helpers for python language related functions."""
+import sys
+from collections import namedtuple
+from typing import NamedTuple, Iterable, Any
+
+
+def is_python_more_recent(minor: int) -> bool:
+    """States whether the python in use is more recent than set minor version."""
+    return sys.version_info.minor >= minor
+
+
+def named_tuple_with_defaults(typename: str, field_names: Iterable[str], defaults: Iterable[Any]) -> NamedTuple:
+    """Gets a named tuple set with default values.
+
+    defaults field only appeared in Python 3.7.
+    See https://stackoverflow.com/questions/11351032/namedtuple-and-default-values-for-optional-keyword-arguments
+    """
+    if is_python_more_recent(7):
+        return namedtuple(typename=typename, field_names=field_names, defaults=defaults)  # type:ignore
+    else:
+        tuple: NamedTuple = namedtuple(typename=typename, field_names=field_names)  # type:ignore
```

Comment:
I wish we could use `dataclasses` instead of `NamedTuples`. Unfortunately they're only part of the stdlib in python 3.7+